### PR TITLE
DOC: Fix troubleshooting code snippet when env vars are empty

### PR DESCRIPTION
**Pull request by seberg**
The previous snippet worked fine, but not when the environment
variables were completly empty.

---

Maybe someone has a shorter idea?

### CODE REVIEW

File: doc/source/user/troubleshooting-importerror.rst [hunk 1]
```diff
@@ -133,10 +133,17 @@ your system. If you can open a correct python shell, you can also run the
...
+    if PYTHONPATH is None:
+        print("The PYTHONPATH is not used.")
+    else:
+        print("The PYTHONPATH includes:", PYTHONPATH.split(os.pathsep))
+
+    PATH = os.environ.get('PATH', None)
+    if PATH is None:
+        print("The PATH is not used.")
+    else:
+        print("The PATH includes:", PATH.split(os.pathsep))
```
Comment (rossbar) [inline doc/source/user/troubleshooting-importerror.rst:146]:
```suggestion
    print("PYTHONPATH: ", os.environ.get('PYTHONPATH'))
    print("PATH: ", os.environ.get('PATH'))
```
A slightly shorter suggestion without the nice "{} is not used" message. Alternatively, we could suggestion running these in an interpreter, though I suppose having it written out in this form is nicer for those who might want to run it as a script to introspect the setup of Python embedded in an IDE etc.

Comment (seberg) [reply doc/source/user/troubleshooting-importerror.rst:146]: Yeah, if I don't mind `['']` being printed for an empty path (which I guess is fine), can even add the splitting to make it easier for some users to read...

**Review by rossbar — COMMENTED** [review]
The changes to the troubleshooting doc look good to me. It looks like an unrelated change in `numpy/core/multiarray.py` might have snuck in here - did you intend to include that in this PR?

**Comment by seberg** [comment]
ops, no, that was a random try (although its a bit prettier :)).

**Review by rossbar — APPROVED** [review]
LGTM

File: doc/source/user/troubleshooting-importerror.rst [hunk 2]
```diff
@@ -133,10 +133,10 @@ your system. If you can open a correct python shell, you can also run the
...
 
     import os
-    PYTHONPATH = os.environ['PYTHONPATH'].split(os.pathsep)
-    print("The PYTHONPATH is:", PYTHONPATH)
-    PATH = os.environ['PATH'].split(os.pathsep)
-    print("The PATH is:", PATH)
+    paths = os.environ.get('PYTHONPATH')
+    print("PYTHONPATH:", paths.split(os.pathsep) if paths else "empty")
+    paths = os.environ.get('PATH')
+    print("PATH:", paths.split(os.pathsep) if paths else "empty")
```
Comment (eric-wieser) [inline doc/source/user/troubleshooting-importerror.rst:139]: Note that on Windows this is quite unfriendly, as all the parts contain backslashes, and printing a `list` shows the repr complete with escape sequences

Comment (seberg) [reply doc/source/user/troubleshooting-importerror.rst:139]: Eeek. So `", ".join` or just print directly?

Comment (seberg) [reply doc/source/user/troubleshooting-importerror.rst]: nvm. lets just print them directly and if someone complains its confusing we can try to be smart about it...

Comment (eric-wieser) [reply doc/source/user/troubleshooting-importerror.rst]: `"\n".join` might be reasonable. Of course, then it becomes hard to spot empty strings in the path (which mean cwd)

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**